consensus/core/src/core_thread.rs

```diff
@@ -17,6 +18,15 @@ use crate::{
 
 const CORE_THREAD_COMMANDS_CHANNEL_SIZE: usize = 32;
 
+#[async_trait]
+pub(crate) trait CoreThreadDispatcherInterface: Sync + Send + 'static {
```

Comment:
I wonder if its worthwhile adding a comment explaining that this interface was added to enable unit testing?